vision/standard_object/odlc_classify_shape.py

```diff
@@ -69,420 +89,339 @@ def process_shapes(
+        An array of rectangular (cartesian) coordinates
 
     Returns
     -------
-    polygonal_shape : chars.ODLCShape | None
-        Will return one of TRIANGLE, SQUARE, RECTANGLE, TRAPEZOID, PENTAGON, HEXAGON, OCTAGON,
-        STAR, or CROSS from ODLCShape Enum or None if shape does not match any of these.
-    """
-    lengths: npt.NDArray[npt.Shape["*"], npt.Float64] = get_lengths(approx)
-    has_eq_side_lengths: bool = compare_side_len_eq(lengths)
-    good_aspect_ratio: bool = filtering.test_min_area_box(approx, SHAPE_ASPECT_RATIO_RANGE)
-
-    angles: npt.NDArray[npt.Shape["*"], npt.Float64]
-    shape: chars.ODLCShape | None = None
-    match len(approx):
-        case 3:
-            shape = chars.ODLCShape.TRIANGLE
-        case 4:
-            angles = get_angles(approx)
-            # if all angles approximately 90 deg, square or rectangle, else trapezoid
-            if np.all(compare_angles(angles, 90, RIGHT_ANG_THRESH)):
-                # if all side lengths are (approximately) equal then square, else rectangle
-                if has_eq_side_lengths and good_aspect_ratio:
-                    shape = chars.ODLCShape.SQUARE
-                else:
-                    shape = chars.ODLCShape.RECTANGLE
-            else:
-                shape = chars.ODLCShape.TRAPEZOID
-        case 5:
-            if good_aspect_ratio:
-                shape = chars.ODLCShape.PENTAGON
-        case 6:
-            if good_aspect_ratio:
-                shape = chars.ODLCShape.HEXAGON
-        case 7:
-            if good_aspect_ratio:
-                shape = chars.ODLCShape.HEPTAGON
-        case 8:
-            if good_aspect_ratio:
-                shape = chars.ODLCShape.OCTAGON
-        case 10:
-            if has_eq_side_lengths:
-                shape = chars.ODLCShape.STAR
-        case 12:
-            angles = get_angles(approx)
-            # a plus shape should have exactly 8 right angles *inside* of it (2 on each bar)
-            if np.count_nonzero(compare_angles(angles, 90, RIGHT_ANG_THRESH)) == 8:
-                shape = chars.ODLCShape.CROSS
-        case _:
-            shape = None
-
-    return shape
-
-
-def compare_side_len_eq(lengs: npt.NDArray[npt.Shape["*"], npt.Float64]) -> bool:
-    """
-    Checks if all of the side lengths given are (approximately) equal.
-
-    Parameters
-    ----------
-    lengths : npt.NDArray[npt.Shape["*"], npt.Float64]
-        An array of all the side lengths of a polygon (from get_lengths())
-
-    Returns
-    -------
-    eq_side_lengths : bool
-        True if all of the given side lengths are approximately equal
+    shape : chars.ODLCShape | None
+        Returns an array of polar coordinates
     """
-    return bool(
-        np.all(np.where(abs((lengs / np.mean(lengs)) - 1) < SIDE_LEN_EQ_THRESH, True, False))
-    )
+    polar: NDArray[Shape["*, 2"], Float64] = []
+    # Stores an array of angles and radii as tuples (radius, angle)
+    for i in range(len(arr)):
```

Comment:
This can be done with numpy functions instead of a for loop, which should be faster.
This could mean that `cartesian_array_to_polar` and `cartesian_to_polar` could be combined into a single function that just acts on a numpy array.